ui/discovery/ssdp.go

```diff
@@ -84,6 +84,7 @@ func (ss *ssdpServer) Start() (err error) {
 		return errors.Wrap(err, "failed to serve device description document")
 	}
 
+	// TODO: this should be initialized ahead of time, not on start as this will cause races and forces nil checks.
```

Comment:
I suggest to remove this TODO. ssdpServer is a wrapper around ssdp lib so this looks like a best approach to keep it as is. You just need to handle Stop correctly as you did.